api/coreservice.go

```diff
@@ -1731,3 +1733,45 @@ func (core *coreService) Track(ctx context.Context, start time.Time, method stri
 		Success:      success,
 	}, size)
 }
+
+func (core *coreService) traceTx(ctx context.Context, txctx *tracers.Context, config *tracers.TraceConfig, simulateFn func(ctx context.Context) ([]byte, *action.Receipt, error)) ([]byte, *action.Receipt, any, error) {
+	var (
+		tracer vm.EVMLogger
+		err    error
+	)
+	switch {
+	case config == nil:
+		tracer = logger.NewStructLogger(nil)
+	case config.Tracer != nil:
+		// Define a meaningful timeout of a single transaction trace
+		timeout := defaultTraceTimeout
+		if config.Timeout != nil {
+			if timeout, err = time.ParseDuration(*config.Timeout); err != nil {
+				return nil, nil, nil, err
+			}
+		}
+		t, err := tracers.New(*config.Tracer, new(tracers.Context))
+		if err != nil {
+			return nil, nil, nil, err
+		}
+		deadlineCtx, cancel := context.WithTimeout(ctx, timeout)
```

Comment:
should we set `ctx = deadlineCtx` to pass it into evm?

Reply:
no need, just call `t.Stop()` to stop execution if timeout